frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -332,8 +354,46 @@ export const experimentLogic = kea<experimentLogicType>({
         updateExperimentSuccess: async ({ experimentData }) => {
             actions.updateExperiments(experimentData)
         },
+        setNewExperimentData: async ({ experimentData }, breakpoint) => {
+            const experimentEntitiesChanged =
```

Comment:
@neilkakkar you're the best but this is some python style code written in javascript instead LOL. It's not super clear to me what is happening here unless I thoroughly read through the entire function

Reply:
It definitely works though just not the easiest to parse

Reply:
Do you have any recommendations on how to make it clean? 😅

Reply:
as examples I would bump `if (!experimentData.filters || Object.keys(experimentData.filters).length === 0) {` to the top, and probably just split up the display errors code under an action/listener that's named more explicitly, though that's probably what kea forms error handling will help with lol :P